utils/fmp_data_cleaner.py

```diff
@@ -0,0 +1,193 @@
+                       VALUES (?, ?, ?, DATE('now'))""",
+                    (symbol, reason, "PRICE_HOLE")
+                )
+                inserted_count += cursor.rowcount
+                
+            conn.commit()
+            
+        return inserted_count
+    
+    def clean_unaccounted_splits(self, dry_run: bool, multiple_threshold: float) -> None:
+        """
+        Main method to detect and optionally clean unaccounted splits.
+        
+        Args:
+            dry_run: If True, only report issues without writing to database
+            multiple_threshold: Minimum change in volume/price to flag as potential split
+        """
+        self.logger.info(f"Starting unaccounted split detection (dry_run={dry_run})")
+        
+        suspicious_symbols = self.detect_unaccounted_splits(multiple_threshold)
+        
+        if not suspicious_symbols:
+            self.logger.info("No suspicious symbols found")
+            return
+            
+        self.logger.info(f"Found {len(suspicious_symbols)} suspicious symbols:")
+            
+        if not dry_run:
+            inserted_count = self.write_suspicious_symbols(suspicious_symbols)
+            self.logger.info(f"Inserted {inserted_count} suspicious symbols into database")
+        else:
+            with open(self.bad_data_file, 'w') as f:
+                for symbol, reason in suspicious_symbols.items():
+                    f.write(f"{symbol} | {self.symbols_info[symbol]['exchange_short_name']} | {self.symbols_info[symbol]['sector']} | {self.symbols_info[symbol]['industry']}: {reason}\n")
+            self.logger.info("Dry run mode - no records were written to database")
+
+def main():
+    parser = argparse.ArgumentParser(description='Clean FMP data quality issues')
+    parser.add_argument('--no-dry-run', action='store_true',
+                       help='Actually write data (default is dry-run mode)')
+    parser.add_argument('--multiple-threshold', type=float, default=0.8,
+                       help='Minimum change in multiple to flag as suspicious (default: 0.8)')
+    parser.add_argument('--db-path', type=str, default=FMP_DB_PATH,
+                       help='Path to the FMP database')
+    parser.add_argument('--log-level', type=str, default='INFO',
+                       choices=['DEBUG', 'INFO', 'WARNING', 'ERROR'],
+                       help='Logging level')
+    
+    args = parser.parse_args()
+    
+    # Setup logging
+    logging.basicConfig(
+        level=getattr(logging, args.log_level),
+        format='%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+    )
+    
+    # Create cleaner and run
+    cleaner = FMPDataCleaner(args.db_path)
+    cleaner.clean_unaccounted_splits(
+        dry_run=not args.no_dry_run,  # Default is dry-run unless --no-dry-run is specified
+        multiple_threshold=args.multiple_threshold
+    )
+
+if __name__ == '__main__':
+    main()
```

Comment:
nit: refactor this into a separate file for cmdline tool and leave the rest as reusable module.